rmw_opensplice_cpp/src/types.cpp

```diff
@@ -27,36 +31,120 @@ create_type_name(
          "::" + sep + "::dds_::" + callbacks->message_name + "_";
 }
 
+CustomDataReaderListener::CustomDataReaderListener()
+: print_discovery_logging_(false)
+{
+  // TODO(wjwwood): replace this with actual console logging when we have that
+  char * discovery_logging_value = nullptr;
+  const char * discovery_logging_env = "RMW_PRINT_DISCOVERY_LOGGING";
+  size_t discovery_logging_size;
+#ifndef _WIN32
+  discovery_logging_value = getenv(discovery_logging_env);
+  if (discovery_logging_value) {
+    discovery_logging_size = strlen(discovery_logging_value);
+  }
+#else
+  _dupenv_s(&discovery_logging_value, &discovery_logging_size, discovery_logging_env);
+#endif
+  if (discovery_logging_value) {
+    std::string str(discovery_logging_value, discovery_logging_size);
+    std::string str_lower(str);
+    std::transform(str_lower.begin(), str_lower.end(), str_lower.begin(), [](unsigned char c) {
+      return std::tolower(c);
+    });
+    if (str != "0" && str_lower != "false" && str_lower != "off") {
+      print_discovery_logging_ = true;
+    }
+#ifdef _WIN32
+    free(discovery_logging_value);
+#endif
+  }
+}
+
+size_t
+CustomDataReaderListener::count_topic(const char * topic_name)
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+  auto it = topic_names_and_types_.find(topic_name);
+  if (it == topic_names_and_types_.end()) {
+    return 0;
+  } else {
```

Comment:
nitpick, you can omit the "else" and just `return it->second.size();` at the end of this function.